src/pages/workspaces/workspace/WorkspaceContainer.js

```diff
@@ -164,15 +164,19 @@ export const WorkspaceContainer = ({
   const [leavingWorkspace, setLeavingWorkspace] = useState(false);
   const workspaceLoaded = !!workspace;
   const isGoogleWorkspaceSyncing = workspaceLoaded && isGoogleWorkspace(workspace) && workspace.workspaceInitialized === false;
+  const [workspaceSyncingChecks, setWorkspaceSyncingChecks] = useState(0);
 
   useEffect(() => {
     if (isGoogleWorkspaceSyncing) {
-      Ajax().Metrics.captureEvent(Events.permissionsSynchronizationDelayDisplayed, {
-        accessLevel: workspace.accessLevel,
-        createdDate: workspace.workspace.createdDate,
-        isWorkspaceCreator: workspace.workspace.createdBy === getUser().email,
-        ...extractWorkspaceDetails(workspace),
-      });
+      setWorkspaceSyncingChecks(workspaceSyncingChecks + 1);
+      if (workspaceSyncingChecks >= 2) {
+        Ajax().Metrics.captureEvent(Events.permissionsSynchronizationDelay, {
```

Comment:
I don't understand the point of doing this-- the event is just a metric about whether the user is setting the message that they don't have access. It has no interaction with FastPass itself.

Also, doing an increment sort of thing here doesn't make sense. This is an effect that is only called when the value of `isGoogleWorkspaceSyncing` changes.

Reply:
One possible solution would be to set a timeout for a given time interval, and then emit the event if `isGoogleWorkspaceSyncing` is still true when the timeout fires. But I think you might need to rethink this event in general-- what is it that you are trying to capture at this point? How many users see the message, or the length of time that they see the message?

Reply:
What's happening here is that UI fires off a `checkBucketReadAccess` request to Rawls, and if Rawls finds an issue with the access, it'll issue a FastPass grant. It'll still return an error to the UI, since permissions propagations aren't immediate. The UI then waits some amount of time, and calls `checkBucketReadAccess` again. Rawls does the checks, and if everything checks out, returns a success. We only want to alert if the second call to `checkBucketReadAccess` fails, so that the FastPass has time to take effect.

I could see Rawls returning a success if it vends a FastPass, but that doesn't _guarantee_ that the permissions are fixed. It can still take "sometimes 7 minutes or longer" for a IAM permissions to propagate.

Reply:
But, I do see that its looking at `isGoogleWorkspaceSyncing` changing, so yeah, its not having the intended effect!